apps/examples/testcase/ta_tc/wifi_manager/itc/itc_wifi_manager_main.c

```diff
@@ -303,6 +319,436 @@ static void itc_wifi_manager_scan_ap_p(void)
+		ret = wifi_manager_init(&wifi_callbacks);
+		if (ret == WIFI_MANAGER_SUCCESS) {
+			s_init_cnt++;
+		} else {
+			init_cnt++;
+			continue;
+		}
+		init_cnt++;
+
+		ret = wifi_manager_set_mode(STA_MODE, NULL);
+		TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+		ret = wifi_manager_connect_ap(&config);
+		if (ret == WIFI_MANAGER_SUCCESS) {
+			s_join_cnt++;
+		}
+		join_cnt++;
+		WIFITEST_WAIT;
+		sleep(5);
+		ret = wifi_manager_disconnect_ap();
+
+		if (ret != WIFI_MANAGER_SUCCESS) {
+			printf("Wifi disconnect fails\n");
+		} else {
+			s_leave_cnt++;
+			WIFITEST_WAIT;
+		}
+		leave_cnt++;
+		sleep(5);
+		ret = wifi_manager_deinit();
+		if (ret == WIFI_MANAGER_SUCCESS) {
+			s_deinit_cnt++;
+		}
+		deinit_cnt++;
+	}
+
+	printf("\nSuccess Ratio \n");
+	printf("[Initialization Wifi] Success Ratio = %.2f%\n", (s_init_cnt / init_cnt) * 100);
+	printf("[Connect Wifi]        Success Ratio = %.2f%\n", (s_join_cnt / join_cnt) * 100);
+	printf("[Disconnect Wifi]     Success Ratio = %.2f%\n", (s_leave_cnt / leave_cnt) * 100);
+	printf("[Deinitization Wifi]  Success Ratio = %.2f%\n", (s_deinit_cnt / deinit_cnt) * 100);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @testcase         itc_wifi_manager_average_stoping_ap
+ * @brief            Looking average time taken while deinit wifi
+ * @scenario
+ * @apicovered
+ * @precondition     none
+ * @postcondition    none
+ */
+
+static void itc_wifi_manager_average_stoping_ap(void)
+{
+	int i = 0;
+	double average = 0.0f;
+	struct timeval start_t;
+	struct timeval end_t;
+	g_connect_status = false;
+	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
+
+	wifi_manager_ap_config_s config;
+	config.ssid_length = strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_SSID);
+	config.passphrase_length = strlen(CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_PASSPHRASE);
+	strncpy(config.ssid, CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_SSID, config.ssid_length + 1);
+	strncpy(config.passphrase, CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_PASSPHRASE, config.passphrase_length + 1);
+	config.ap_auth_type = (wifi_manager_ap_auth_type_e)CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_AUTHENTICATION;
+	config.ap_crypto_type = (wifi_manager_ap_crypto_type_e)CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_ITC_CRYPTO;
+	printf("AP config: %s(%d), %s(%d), %d %d\n", config.ssid, config.ssid_length, config.passphrase, \
+			config.passphrase_length, config.ap_auth_type, config.ap_crypto_type);
+
+	for (i = 1; i <= LOOP_SIZE; i++) {
+		ret = wifi_manager_init(&wifi_callbacks);
+		TC_ASSERT_EQ("wifi_manager_init", ret, WIFI_MANAGER_SUCCESS);
+
+		ret = wifi_manager_set_mode(STA_MODE, NULL);
+		TC_ASSERT_EQ("wifi_manager_set_mode", ret, WIFI_MANAGER_SUCCESS);
+
+		ret = wifi_manager_connect_ap(&config);
+		TC_ASSERT_EQ("wifi_manager_connect_ap", ret, WIFI_MANAGER_SUCCESS);
+		WIFITEST_WAIT;
+		sleep(5);
+
+		ret = wifi_manager_disconnect_ap();
+		TC_ASSERT_EQ("wifi_manager_disconnect_ap", ret, WIFI_MANAGER_SUCCESS);
+		WIFITEST_WAIT;
+		sleep(5);
+
```

Comment:
Since we measure only deinit time, from a STA mode, is it necessary to connect and disconnect inside the loop? Would this affect the measurement?

Reply:
Yes, it will be good if we do connect and disconnect inside the loop to measure actual performance.